docs/manual/src/docs/asciidoc/_includes/servlet/oauth2/oauth2-resourceserver.adoc

```diff
@@ -3038,7 +3038,7 @@ Additionally, it is published as an `AuthenticationFailureBadCredentialsEvent`,
 @Component
 public class FailureEvents {
 	@EventListener
-    public void onFailure(AuthenticationFailureEvent failure) {
+    public void onFailure(AuthenticationFailureBadCredentialsEvent failure) {
```

Comment:
While you are at it, it would be nice if we fix the name of the parameter, from `failure` to `badCredentials`.  This way aligning with the next line.

Reply:
Thank you for spotting this! I have added the changes.